### PR TITLE
Calculator UI

### PR DESCRIPTION
**Pull request by Sboursen**
# Description

This is the second milestone of the **Math Magicians** project.

## Milestone requirements

- [x] Delete the boilerplate from CRA (text, images, styles).

- [x] Create a directory called components.

- [x] create a new Calculator.js file.

- [x] Follow the design spesifications

- [x] Use class-based components.

- [x] App should render without a problem

## General requirements

- [x] There are no linter errors.
- [x] Followed the **Gitflow**.
- [x] The documentation is professional.

## JavaScript requirements

- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md)

## Additional notes

image

Thank you for taking the time to review this pull request.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,34 @@
 import logo from './logo.svg';
 import './App.css';
+import CalculatorBody from './components/calculatorBody';
 
 function App() {
   return (
     <div className="App">
       <header className="App-header">
```
Comment (julie-ify) [inline src/App.js]: - [ ] The `App.js` component should be class-based. Kindly make use of a class component as it is part of the project requirement.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Sboursen,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

- [x] **[REQUIRED]** Kindly remove all the boilerplate that got created by the CRA (texts, images, and styles). Check the public folder of your application and remove all the boilerplate leaving the index.hml file.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (Sboursen) [reply src/App.js]: @julie-ify 
I already used class-based react components for the calculator and the button
There is no requirement saying that I should **Only** use class-based component

**Comment by Sboursen** [comment]
@julie-ify 
Thank you for the review,
I left some questions that I hope you could answer
Best,

File: src/App.js (hunk 1 above)
Comment (Emmyn5600) [reply src/App.js]: @Sboursen, A class-based should look like this:

```javascript
class App extends Component {
  render() {
    return {
      <CalculatorBody />
    };
  }
}
```

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Sboursen  

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (Sboursen) [reply src/App.js]: Hi @Emmyn5600  @julie-ify,

Thank you for the suggestion, but I already know how to refractor a function-based component into a class-based one!
**My point is**, there is no requirement that says I should only use class-based components
Please check out the project requirements here [link](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md)

Comment (Sboursen) [reply src/App.js]: I already used class-based components multiple times
please check:
- [here](https://github.com/Sboursen/math-magicians/pull/2/files#diff-974f77c88a7586d7b16ad8950fb86c80e2eb377667571b0c3286014639a62c47R5-R23)
- [here](https://github.com/Sboursen/math-magicians/pull/2/files#diff-8b64f45cfa459d87c79a27ec59c4823a0002d6d062bcd9bbea4caad219786facR6-R43)

Comment (Emmyn5600) [reply src/App.js]: @Sboursen , in the requirements they mentioned to `Use class based components.` so it is easy you can refactor the App.js file with the class-based component to move forward as well as to move to the next milestone.

Comment (julie-ify) [reply src/App.js]: Hi @Sboursen,

Kindly know that it is required in the code review guideline and I also checked the project requirement link you attached and saw this `Use class based components.`

**Comment by Sboursen** [comment]
Hi @Emmyn5600 
Thank you for the feedback
However, I believe that your requested change is unnecessary blocking this pull request.
please check the comment above!
Looking forward to your reply

File: src/App.js (hunk 1 above)
Comment (Emmyn5600) [reply src/App.js]: @Sboursen, So, you only have to refactor the App.js.

Comment (Sboursen) [reply src/App.js]: @Emmyn5600 @julie-ify 

I totally understand that. And actually, I did use them like It's shown in the comment above.
My point is that the requirement doesn't say that I need to use Only the class-based components.
As long I'm using them somewhere, it doesn't matter f I use a functional-based ones too.

**Review by elyor-doniyorov — APPROVED** [review]
Hi @Sboursen, 

In my opinion, your project components are ready!  There is nothing else to say other than... well done :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**